### PR TITLE
Updates WAFS according to EE2 standards

### PR DESCRIPTION
**Pull request by YaliMao-NOAA**
https://docs.google.com/document/d/1fYsVrcrjj_NxJBYG6v0sDhi-vYxShT2l2EUauHIByLs/edit?pli=1

## Pull Request Testing ##

- [ ] Describe testing already performed for this Pull Request:</br>

- [ ] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

- [ ] Has the code been checked to ensure that no errors occur during the execution? **[Yes or No]**

- [ ] Do these updates/additions include sufficient testing updates? **[Yes or No]**

- [ ] Please complete this pull request review by **[Fill in date]**.</br>

## Pull Request Checklist ##

- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR description above.
- [ ] Ensure the PR title matches the feature branch name.
- [ ] Check the following:
- [ ]  Instructions provided on how to run
- [ ]  Developer's name is replaced by ${user} where necessary throughout the code
- [ ]  Check that the ecf file has all the proper definitions of variables
- [ ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [ ]  Check to see that the output directory structure is followed
- [ ]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [ ] After submitting the PR, select **Development** issue with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue.

### CODE REVIEW

**Comment by PerryShafran-NOAA** [comment]
@AliciaBentley-NOAA @ShelleyMelchior-NOAA @YaliMao-NOAA This update includes the wafs plots. Should these be removed?  I know that we want to keep develop up to date with everything, but the merge will put plots back into the parallel.

**Comment by AliciaBentley-NOAA** [comment]
Hi Perry,

We still want the WAFS plots in the develop branch since Yali will run
these in dev. Please make sure to manually delete them from the emc.vpppg
parallel. I don't think that it's too many scripts. Thanks!

Take care,
Alicia
___________________________________
*Alicia M. Bentley, Ph.D.*
Global Verification Project Lead
Model Evaluation Group (MEG) <https://www.emc.ncep.noaa.gov/users/meg/home/>
NOAA/NWS/NCEP/EMC (VPPPG Branch)
Work Phone: ‪(301) 683-3783‬

**Comment by AliciaBentley-NOAA** [comment]
@YaliMao-NOAA Thank you very much for submitting your EE2 PR. I see just a couple of things that you need to change before Perry or Shelley can test it. Please make these changes and let us know when they are updated in this PR:

1) NCO would like SENDMAIL to be turned off for their code delivery. Can you also add the line "export SENDMAIL=${SENDMAIL:-**NO**}" into your j-jobs? I think the dev/drivers/scripts being set to yes should override what's in your j-job, but we want SENDMAIL to be set to NO in the J-jobs when NCO runs things.

2) We only asked you to delete the lines with "**export KEEPDATA=**" in EVS. Can you please add the following "if statements" with KEEPDATA back into your j-jobs? We will need these if statements for EVS to work. Thanks!

> if [ "$KEEPDATA" != "YES" ] ; then
>   rm -rf $DATA
> fi

@ShelleyMelchior-NOAA @PerryShafran-NOAA

**Comment by AliciaBentley-NOAA** [comment]
@YaliMao-NOAA I see you already added the KEEPDATA if statements back, which is great.

@PerryShafran-NOAA -  I do not see the following SEND____ exports in the WAFS j-jobs:

> ####################################
> export SENDECF=${SENDECF:-YES}
> export SENDCOM=${SENDCOM:-YES}
> export SENDDBN=${SENDDBN:-NO}
> export SENDMAIL=${SENDMAIL:-NO}

Can you please make sure these get in the j-jobs so that WAFS can be run in ops? Thanks!

**Comment by YaliMao-NOAA** [comment]
Should the following lines be included in jobs/ too?

I can add them to j-jobs.

> > ####################################
> > export SENDECF=${SENDECF:-YES}
> > export SENDCOM=${SENDCOM:-YES}
> > export SENDDBN=${SENDDBN:-NO}
> > export SENDMAIL=${SENDMAIL:-NO}

**Comment by YaliMao-NOAA** [comment]
The EE2 documents should be updated to export SENDMAIL=${SENDMAIL:-NO}. It is listed as YES. I am going to change from YES to NO

**Comment by PerryShafran-NOAA** [comment]
Yes, these lines should be in your J-job.

**Comment by AliciaBentley-NOAA** [comment]
@YaliMao-NOAA Putting SENDMAIL=${SENDMAIL:-YES} in your dev/drivers/scripts was correct. That is what the EE2 documents asked for, so please leave that there. We want SENDMAIL to be set to YES in your dev code. 

NCO doesn't want to get any emails from us in ops, so a default line for SENDMAIL should be added to the j-jobs that says SENDMAIL=${SENDMAIL:-NO}. What is in your dev drivers overrides what is in the j-jobs. With these settings, SENDMAIL will be set to YES in dev and sent to NO when NCO runs it. Does that make sense?

**Comment by YaliMao-NOAA** [comment]
@AliciaBentley-NOAA Got you! Going to make the changes!

File: dev/drivers/scripts/wafs/stats/jevs_wafs_atmos_stats.sh [hunk 1]
```diff
@@ -23,6 +23,14 @@ export COMPONENT=wafs
...
 export VERIF_CASE=grid2grid
 
+############################################################
+Set data directives
+############################################################
+export SENDECF=${SENDECF:-YES}
+export SENDCOM=${SENDCOM:-YES}
+export SENDDBN=${SENDDBN:-NO}
+export SENDMAIL=${SENDMAIL:-NO}
+
```
Comment (PerryShafran-NOAA) [inline dev/drivers/scripts/wafs/stats/jevs_wafs_atmos_stats.sh]: @YaliMao-NOAA These four lines need to be in the J-job, not the driver script.  Place these lines in JEFS_WAFS_ATMOS_STATS and JEVS_WAFS_ATMOS_PLOTS, and remove them from here.

Here's an example, from the current branch I am working in:

https://github.com/PerryShafran-NOAA/EVS/blob/bugfix/analyses_EE2/jobs/JEVS_ANALYSES_STATS

Thanks!

Perry

**Comment by AliciaBentley-NOAA** [comment]
@YaliMao-NOAA Can you also update "export KEEPDATA=YES" on Line 54 of dev/drivers/scripts/wafs/stats/jevs_wafs_atmos_stats.sh? This was also on the EE2 checklist. Thank you!

**Comment by AliciaBentley-NOAA** [comment]
@YaliMao-NOAA Thanks! I think this EE2 PR is ready to test now, unless @PerryShafran-NOAA or @ShelleyMelchior-NOAA think otherwise!

File: jobs/JEVS_WAFS_ATMOS_PLOTS [hunk 2]
```diff
@@ -60,7 +68,7 @@ export COMINstat=$COMIN/stats/$COMPONENT  # will be appended with wafs.$VDATE or
 
 # COMOUT folder
 # evs_ver defined in versions/run.ver
-export COMOUT=${COMOUT:-$COMIN/$STEP/$COMPONENT/$RUN.$VDATE}
+export COMOUT=${COMOUT:-$(compath.py ${envir}/com/${NET}/${evs_ver})/$STEP/$COMPONENT/$RUN.$VDATE}
```
Comment (ShelleyMelchior-NOAA) [inline jobs/JEVS_WAFS_ATMOS_PLOTS]: @YaliMao-NOAA , this is close! Please change to:
`export COMOUT=${COMOUT:-$(compath.py -o ${NET}/${evs_ver})/$STEP/$COMPONENT/$RUN.$VDATE}`

**Comment by ShelleyMelchior-NOAA** [comment]
@YaliMao-NOAA , this update is _**IMPERATIVE**_. Please do a sync fork. Your branch `update09282023` is behind develop. I won't begin any testing until your branch is updated. Thanks!

File: jobs/JEVS_WAFS_ATMOS_PLOTS (hunk 2 above)
Comment (YaliMao-NOAA) [reply jobs/JEVS_WAFS_ATMOS_PLOTS]: @ShelleyMelchior-NOAA 
Should [jobs/JEVS_WAFS_ATMOS_STATS](https://github.com/NOAA-EMC/EVS/pull/268/files/b0f930b0e6a33c11bf2870ee18a32abcaf5bd28e#diff-b95b15e3286cd3bdbfe03ed8388932e3988ebe5baf5e32e81659d13a9dead702) have this change?
export COMOUT=${COMOUT:-$(compath.py -o ${NET}/${evs_ver})}

Comment (ShelleyMelchior-NOAA) [reply jobs/JEVS_WAFS_ATMOS_PLOTS]: Yes! You can make that change in JEVS_WAFS_ATMOS_STATS, too. Good catch!

**Comment by YaliMao-NOAA** [comment]
@ShelleyMelchior-NOAA I made the changes to COMOUT and merged EVS develop branch.

**Comment by ShelleyMelchior-NOAA** [comment]
I have submitted wafs stats processing.
```
83115493.cbqs01                shelley.melchi* dev             jevs_wafs_stats   131194    1     4   10gb 01:00 R 00:00
```
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_wafs_stats.o83115493`

File: dev/drivers/scripts/wafs/plots/jevs_wafs_atmos_plots.sh
Comment (ShelleyMelchior-NOAA) [inline dev/drivers/scripts/wafs/plots/jevs_wafs_atmos_plots.sh:1]: I know plots will not be in ops code, but I wanted to run it anyway. I see one thing that needs to be fixed.  This line needs to be removed:
```
export FIXevs=/lfs/h2/emc/vpppg/noscrub/emc.vpppg/verification/EVS_fix
```
Now we run `sorc/build-fix-slink.sh` to build a symbolic link for $FIXevs and $HOMEevs/fix.

**Comment by ShelleyMelchior-NOAA** [comment]
> I have submitted wafs stats processing.
> 
> ```
> 83115493.cbqs01                shelley.melchi* dev             jevs_wafs_stats   131194    1     4   10gb 01:00 R 00:00
> ```
> 
> logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_wafs_stats.o83115493`

Stats processing is completed. Please review.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/wafs/wafs.20230928`

File: dev/drivers/scripts/wafs/plots/jevs_wafs_atmos_plots.sh
Comment (YaliMao-NOAA) [reply dev/drivers/scripts/wafs/plots/jevs_wafs_atmos_plots.sh:1]: Removed. Thank you @ShelleyMelchior-NOAA

**Comment by YaliMao-NOAA** [comment]
@ShelleyMelchior-NOAA Both logfile and outputs look good to me

**Comment by ShelleyMelchior-NOAA** [comment]
Excellent! Thanks Yali! I am running plots now. It finishes and produces tar files that are a good size. However, there are some nitpicks in the driver script that you can add just for completeness. Let's add the setting of VDATE and COMOUT. 
Add
```
export cyc=${cyc:-00}
export cycle=t${cyc}z
setpdy.sh
. ./PDY
export VDATE=${VDATE:-$PDYm1}
```
anywhere ahead of
```
export COMOUT=/lfs/h2/emc/ptmp/$USER/evs/$evs_ver/${STEP}/${COMPONENT}/${RUN}.${VDATE}
```

**Comment by YaliMao-NOAA** [comment]
@ShelleyMelchior-NOAA 

> setpdy.sh
> . ./PDY
> export VDATE=${VDATE:-$PDYm1}

The lines are in j-jobs. Is it ok? Must they be in dev drivers?

**Comment by ShelleyMelchior-NOAA** [comment]
It can be in there. It is needed in order to define VDATE. VDATE is needed for COMOUT to match how COMOUT is defined in the J-job. But to be clear, please keep it in the J-job, too.

**Comment by ShelleyMelchior-NOAA** [comment]
I finished running wafs plots with the changes discussed above. I made the changes in my local checkout of your branch. But I will look for them to come from you in this PR. Please review.
COMOUT=`/lfs/h2/emc/ptmp/shelley.melchior/evs/v1.0/plots/wafs/atmos.20230928`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_wafs_plots.o83120058`

**Comment by YaliMao-NOAA** [comment]
Yes, they are in this order in WAFS j-jobs.

> export cyc=${cyc:-00}
> export cycle=t${cyc}z
> setpdy.sh
> . ./PDY
> export VDATE=${VDATE:-$PDYm1}

**Comment by YaliMao-NOAA** [comment]
@ShelleyMelchior-NOAA The logfile and plottings look good to me.

**Comment by ShelleyMelchior-NOAA** [comment]
Great! Thanks Yali! I am going to update the plots driver to include the updates I needed to add in order for plots to run from my environment. Then I think this PR is nearly ready for merge!

**Review by ShelleyMelchior-NOAA — APPROVED** [review]
All tests produced expected results. This PR is approved for merge.

**Pull request merged**